ice-adapter/src/main/java/com/faforever/iceadapter/ice/PeerIceModule.java

```diff
@@ -95,8 +95,23 @@ private void createAgent() {
      */
     private void gatherCandidates() {
         log.info(getLogPrefix() + "Gathering ice candidates");
-        GameSession.getIceServers().stream().flatMap(s -> s.getStunAddresses().stream()).map(StunCandidateHarvester::new).forEach(agent::addCandidateHarvester);
-        GameSession.getIceServers().forEach(iceServer ->
+        List<IceServer> iceServers = GameSession.getIceServers();
+        OptionalDouble minRtt = iceServers.stream()
+                .map(server -> server.getRtt().join())
+                .filter(OptionalDouble::isPresent)
+                .mapToDouble(OptionalDouble::getAsDouble)
+                .min();
+        log.info(getLogPrefix() + "minRtt: " + minRtt.toString());
+        // Ignore all servers which had an rtt greater than the minimum
+        if (minRtt.isPresent()) {
+            double minRttAsDouble = minRtt.getAsDouble();
+            iceServers = iceServers.stream().filter(server -> {
+                OptionalDouble rtt = server.getRtt().join();
+                return !(rtt.isPresent() && rtt.getAsDouble() > minRttAsDouble);
+            }).collect(Collectors.toList());
+        }
+        iceServers.stream().flatMap(s -> s.getStunAddresses().stream()).map(StunCandidateHarvester::new).forEach(agent::addCandidateHarvester);
+        iceServers.forEach(iceServer ->
```

Comment:
I'd not select the turn server with the lowest latency. This would put a lot of stress on maybe weaker external servers. I suggest picking the FAF turn server unless its latency is larger than 250? ms, in which case we pick the server with the lowest latency.

Reply:
So how do we know what is an official server and what isnt? Just by checking if `faforever.com` is in the url?

Reply:
I can't think of a better way atm.